frontend/apps/reader/modules/readerdictionary.lua

```diff
@@ -313,14 +304,50 @@ If you'd like to change the order in which dictionaries are queried (and their r
 end
 
 function ReaderDictionary:onLookupWord(word, box, highlight, link)
+    logger.dbg("lookup word:", word, box)
+    -- escape quotes and other funny characters in word
+    word = self:cleanSelection(word)
+    logger.dbg("stripped word:", word)
+
     self.highlight = highlight
+
     -- Wrapped through Trapper, as we may be using Trapper:dismissablePopen() in it
     Trapper:wrap(function()
-        self:stardictLookup(word, box, link)
+        self:stardictLookup(word, self.enabled_dict_names, not self.disable_fuzzy_search, box, link)
     end)
     return true
 end
 
+function ReaderDictionary:onHtmlDictionaryLinkTapped(dictionary, link)
+    if not link.uri then
+        return
+    end
+
+    -- The protocol is either "bword" or there is no protocol, only the word.
+    -- https://github.com/koreader/koreader/issues/3588#issuecomment-357088125
+    local word = link.uri
+    local url_prefix = "bword://"
+    if link.uri:sub(1,url_prefix:len()) == url_prefix then
+        word = link.uri:sub(url_prefix:len() + 1)
+    elseif link.uri:find("://") then
+        return
+    end
+
+    local link_box = Geom:new{
+        x = link.x0,
+        y = link.y0,
+        w = math.abs(link.x1 - link.x0),
+        h = math.abs(link.y1 - link.y0),
+    }
+
+    self.highlight = nil
```

Comment:
No side effect with unsetting `self.highlight` (like the highlight in the book not being unhighlighted when you leave all dict windows) ? (Just reading code, can't test for now.)

Reply:
You're right, it's better to leave the highlight alone and keep the value that was set when the dictionary got opened. I didn't see any side effects though. :)